lib/cosmos/system/target.rb

```diff
@@ -124,7 +124,9 @@ def process_file(filename)
           usage = "#{keyword} <FILENAME>"
           parser.verify_num_parameters(1, 1, usage)
           begin
-            Cosmos.require_file(parameters[0])
+            # Require absolute path to file in target lib folder.  Prevents name
+            # conflicts at the require step
+            Cosmos.require_file(File.join(@dir, 'lib', parameters[0]))
```

Comment:
It is breaking change.  This is how I implemented the "namespacing" support though.   There are really two aspects to namespacing.   

1. Being able to require a file with the same name.  This handles that.
2. Being able to have two classes with the same name.  The changes to allowing filenames or class names in the config files handles that.